---
title: "Analysis of yeast genomes for 26W MGY360"
author: "Morgan Alford, modified from Sarah McClymont"
date: "January 2026"
output: pdf_document
---

# Intro 

For the MGY360 class, we will generate Nextera DNA libraries (from caspofungin- or micafungin-resistant yeast strains) and sequenced them. Our goal is to find the causative mutation for the antifungal resistance and explore the mechanism behind that resistance. 

We will generate x sequencing libraries: y libraries of mutants, z libraries of their parents 

The general plan is to:
	1) Perform quality control on the sequencing results 
	2) Align to a reference genome 
	3) Perform quality control on the alignment 
	4) Identify variants in mutant and parent strains compared to the reference 
	5) Filter the variants for presence/absence in parent vs mutant
	6) Manually curate the list of variants to identify the most likely causative mutations 
	7) Perform literature searches to explore how these variants could be causing our phenotype of interest
	
# The files 

```{r, eval = F}

# Files are located in /UofT/2026W/MGY360/2-Labs/DryLabs/SequencingData/Mutants
# Files: 
# 

```

# 1) Pre-alignment QC

Use FastQC to check library quality. Summarize FastQC reports with MultiQC and check the generated HTML for sequencing quality issues.

```{r, eval = F}

# batch script: fastqc.sh

#!/bin/bash
#SBATCH --job-name=fastqc
#SBATCH -N 1
#SBATCH -n 10

# FastQC v0.11.9
fastqc -t 10 *.fastq.gz

# Generate report with MultiQC
## pip3 install --user multiqc #v1.13
multiqc .

# Clean up files
mkdir FastQC
mv *fastqc* FastQC

exit 0

```

Output: Individual FASTQC reports. Share with class on GitHub.
Output: MultiQC report summary (fastQC_multiqc_report.html). Look at plots in the HTML to assess overall quality. 

**Conclusions:** Look good in general. 

Quality across reads: Uniformly high. Drops into "yellow" at ends of all reads, but expected with 150bp read length. 
Quality averaged per read: High. Most 30+
Sequence bias: As always, all samples have sequence composition bias at the start of reads, as per usual for Tn5. Nothing of concern here.
GC content: Correct for yeast genome average.
N content: None to speak of. Good sequencing run. No dead flow cell spots.
Seq length: All 150, as expected
Seq duplication: Generally pretty good. A few libraries are going to lose a fair chunk of their reads to this, but should still have enough to call vars
Adapter content: Only a few managed to stay in the green zone. Everyone pretty much has approx 10-30% of their reads ending in adapters. Probably cut the DNA too short. Could consider in future iterations of class decreasing transposase time, increasing DNA input, or titrating Tn5 concentration to prevent tagmentation below 100bp. 

**Samples that are a little wonky:**
B04 - Weird GC content (%GC = 34% instead of 40%), very high duplication rate (20% of sequences are present 10-100 times)
C04: Low sequencing depth. Just under 400k reads instead of 3M+. Primer contam? B10 is also a little low (800k reads)

--> Move forward with mapping! No need to trim reads or adapters, just one library of concern to keep our eye on (B04).

# 2) Align to yeast genome

Use Bowtie2 to align to yeast genome.  
Deduplicate with Picard Tools. 
Index with SAMtools. 

Summarize reports with MultiQC.   

```{r, eval = F}

# alignment.sh

#!/bin/bash
#SBATCH --job-name=alignment
#SBATCH -N 1
#SBATCH -n 12

# Define path to genome indices, downloaded from Bowtie2 website, Feb 2024
  genome_index=/home/ltri/santoslab/smcclymo/MGY360/R64-1-1/R64-1-1 

  
for file in *R1_001.fastq.gz; do

# Make life easier by making the file names simpler and into an extensionlesss variable
# Remove the ends of the names 
	sample=${file%_R1_001.fastq.gz}
# Strip off the first 19 characters
	sample2=${sample:19}

# Write out sample name so can track progress/errors in slurm.out file
  echo $sample2

# Align with Bowtie2 v2.4.2
  # Local alignment to softclip the ends
  # Increase pair distance to 800
  # Write out alignment stats (stderr) to file for multiqc summary
  bowtie2 -p 12 --local -X 800 -x $genome_index -1 $file -2 ${sample}_R2_001.fastq.gz -S ${sample2}.sam 2>${sample2}.align.stats

# Convert to bam and sort with samtools v1.11
  # view -bu turns into uncompressed bam for easy piping into sort
  samtools view -bu ${sample2}.sam | samtools sort -o ${sample2}_sorted.bam

# Deduplicate with picard tools v2.26.11
  picard MarkDuplicates I=${sample2}_sorted.bam O=${sample2}.bam M=${sample2}_markdup.stats

# Index with samtools
  samtools index ${sample2}.bam

# Remove intermediate files
	${sample2}.sam
	${sample2}_sorted.bam
	
done 

# Summarize alignment stats with multiQC 
multiqc . --ignore "FastQC/"

exit 0	

```

Output: BAM and BAI files for each library. BAM files are sorted and duplicates have been flagged. BAIs are indices for visualization later. Figure out a way to upload these to UCSC so everyone has access to their tracks for looking at their variants. Too big for GitHub.
Output: Alignment stats for each library. Upload to GitHub.
Output: MultiQC report (alignment_dups_multiqc_report.html) summarizing the alignment and deduplication stats. Look at this for problem libraries.

% alignment: Pretty much all 95+% - there are 2 libraries of concern. One is B04, which was flagged above. 0% alignment. BLAST'ed some reads from that library - all staph. Another is E09, which has 61.1% alignment. Something funny with that library. Contamination? 

	Investigating: To get the unmapped reads use samtools (flag 4 = unmapped): samtools view -f 4 [file].bam > unmapped_reads_[sample].sam
	Open the SAM file and copy some of the reads to send to BLAST. Not getting any matches. 
	If I allow a lot of mismatches (discontiguous megablast), it actually matches with yeast??? 
	Do notice: There's a LOT of repeats and N's. Sample bad? It looked fine in the FASTQC report, but this is telling a different story.... Not sure what's up. Move forward with the reads we've got, but keep an eye out. 	
	
% Duplication: Very little. All under 5% 

# 3) Check alignment 
	
Look at the mapping quality distribution, fragment length distribution, and genome coverage for each library. 
Use Qualimap BAMQC (v2.3) and summarize with multiqc.

```{r, eval = F}

# qualimap.sh

#!/bin/bash
#SBATCH --job-name=qualimap

for file in *.bam; do
# Use qualimap bamQC tool 
# Make outdir called qualimap and make sure the format is HTML so multiqc can summarize properly. Can't read PDFs

/home/ltri/santoslab/share/software/Qualimap/qualimap_v2.3/qualimap bamQC -bam $file -outdir ${file%.bam}_qualimap -outformat HTML -outfile ${file%.bam}_qualimap.report

done

# Summarize with multiQC
multiqc . --ignore "FastQC/" --ignore "markdup/" --ignore "alignment_stats/"

exit 0

```

Output: PDF reports for each library. Share with class. 

# 4) Call variants

We need to identify all the variants in both the parental and mutant yeast strains. Eventually will compare the list and filter to only look at ones that are in the mutant, but to do that, we need the parental variation too. 

Use bcftools mpileup and bcftools call (v1.9) to get a list of variant sites. 

```{r, eval = F}
 
# This command relies on knowing the ploidy of the yeasts. Samples with 4, 5, 6 in the name are diploid. 
# Move BAMs with *04*, *05*, *06* to a new directory called ploidy2
mv *04*.bam* ploidy2/
mv *05*.bam* ploidy2/  
mv *06*.bam* ploidy2/
# Move the remaining BAMs to a new directory called ploidy1 
mv *.bam* ploidy1/
  
# In the ploidy1 directory, run the following script:   

#!/bin/bash
#SBATCH --job-name=VCF

for file in *.bam; do

bcftools mpileup -f Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa $file -o ${file%.bam}.vcf
bcftools call --ploidy 1 -O v -vc ${file%.bam}.vcf -o ${file%.bam}_vars.vcf

done

exit 0

# In the ploidy2 directory, run the following: 

#!/bin/bash
#SBATCH --job-name=VCF

for file in *.bam; do

bcftools mpileup -f Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa $file -o ${file%.bam}.vcf
bcftools call --ploidy-file ploidy2_file -O v -vc ${file%.bam}.vcf -o ${file%.bam}_vars.vcf

done

exit 0

# Where the contents of teh ploidy2_file are: 
#*  * *     * 2

# Link the VCFs in a folder all together again (ln -s)

# Fix the first column so that it isn't just "I" and "II" but is "chrI" and "chrII." 
for file in *.vcf; do
# Starting at the 49th line, append "chr" to the front of each row
awk 'NR>49 {print "chr"$0}' $file > intermediate.vcf
# Add back in the header 
head -n 48 $file > header.temp
cat header.temp intermediate.vcf > ${file%_vars.vcf}.vcf
done

```

Export the variant calls in VCF format. Upload to GitHub. 
