---
title: "Analysis of yeast genomes for 26W MGY360"
author: "Morgan Alford, modified from Sarah McClymont"
date: "January 2026"
output: pdf_document
---

# Intro 

For the MGY360 class, we will generate Nextera DNA libraries (from caspofungin- or micafungin-resistant yeast strains) and sequenced them. Our goal is to find the causative mutation for the antifungal resistance and explore the mechanism behind that resistance. 

We will generate x sequencing libraries: y libraries of mutants, z libraries of their parents 

The general plan is to:
	1) Perform quality control on the sequencing results 
	2) Align to a reference genome 
	3) Perform quality control on the alignment 
	4) Identify variants in mutant and parent strains compared to the reference 
	5) Filter the variants for presence/absence in parent vs mutant
	6) Manually curate the list of variants to identify the most likely causative mutations 
	7) Perform literature searches to explore how these variants could be causing our phenotype of interest
	
# The files 

```{r, eval = F}

# Files will be located in /home/micbmoe/
# Files: 
# 

```

# 1) Pre-alignment QC

Use FastQC to check library quality. Summarize FastQC reports with MultiQC and check the generated HTML for sequencing quality issues.

```{r, eval = F}

# batch script: fastqc.sh

#!/bin/bash
#SBATCH --job-name=fastqc
#SBATCH -N 1
#SBATCH -n 10

# FastQC v0.12.1
module load StdEnv/2023
module load fastqc/0.12.1
mkdir -p FastQC
fastqc -t 10 -o FastQC *.fastq.gz

# Generate report with MultiQC
module load python/3.13.2
module load gcc arrow/22.0.0
python -m venv multi
pip install multiqc
multiqc FastQC
deactivate

exit 0

```

Output: Individual FASTQC reports. Share with class on GitHub.
Output: MultiQC report summary (fastQC_multiqc_report.html). Look at plots in the HTML to assess overall quality. 

**Conclusions:**  

Quality across reads:  
Quality averaged per read: 
Sequence bias: 
GC content: 
N content: 
Seq length: 
Seq duplication: 
Adapter content:  

**Unusual samples, if any:**

--> Moving forward:

# 2) Align to yeast genome

Use Bowtie2 to align to yeast genome.  
Deduplicate with Picard Tools. 
Index with SAMtools. 

Summarize reports with MultiQC.   

```{r, eval = F}

# script: alignment.sh

#!/bin/bash
#SBATCH --job-name=alignment
#SBATCH -N 1
#SBATCH -n 12

# Define path to genome indices, download from Bowtie2 website
genome_index=
  
for file in *R1_001.fastq.gz; do

# Remove the ends of the names 
	sample=${file%_R1_001.fastq.gz}
# Strip off the first 19 characters
	sample2=${sample:19}

# Write out sample name so can track progress/errors in slurm.out file
  echo $sample2

# Align with Bowtie2 v2.5.4
module load bowtie2/2.5.4

  # Local alignment to softclip the ends
  # Increase pair distance to 800
  # Write out alignment stats (stderr) to file for multiqc summary
  bowtie2 -p 12 --local -X 800 -x $genome_index -1 $file -2 ${sample}_R2_001.fastq.gz -S ${sample2}.sam 2>${sample2}.align.stats

# Convert to bam and sort with samtools v1.22.1
module load samtools/1.22.1
  # view -bu turns into uncompressed bam for easy piping into sort
  samtools view -bu ${sample2}.sam | samtools sort -o ${sample2}_sorted.bam

# Deduplicate with picard tools v3.1.0
module load picard/3.1.0
  picard MarkDuplicates I=${sample2}_sorted.bam O=${sample2}.bam M=${sample2}_markdup.stats

# Index with samtools
  samtools index ${sample2}.bam

# Remove intermediate files
	${sample2}.sam
	${sample2}_sorted.bam
	
done 

# Summarize alignment stats with multiQC 
multiqc . --ignore "FastQC/"

exit 0	

```

Output: BAM and BAI files for each library. BAM files are sorted and duplicates have been flagged. BAIs are indices for visualization later. Figure out a way to upload these to UCSC so everyone has access to their tracks for looking at their variants. Too big for GitHub.
Output: Alignment stats for each library. Upload to GitHub.
Output: MultiQC report (alignment_dups_multiqc_report.html) summarizing the alignment and deduplication stats. Look at this for problem libraries.

**Conclusions:** 
% alignment: 
	Investigating: To get the unmapped reads use samtools (flag 4 = unmapped): samtools view -f 4 [file].bam > unmapped_reads_[sample].sam
	Open the SAM file and copy some of the reads to send to BLAST. 
	Observations:
% duplication: 

# 3) Check alignment 
	
Look at the mapping quality distribution, fragment length distribution, and genome coverage for each library. 
Use Qualimap BAMQC (v2.3) and summarize with multiqc.

```{r, eval = F}

# script: qualimap.sh

#!/bin/bash
#SBATCH --job-name=qualimap
#SBATCH -N 
#SBATCH -n 

module load qualimap/2.3

for file in *.bam; do

# Use qualimap bamQC tool 
# Make outdir called qualimap and make sure the format is HTML so multiqc can summarize properly. Can't read PDFs.

qualimap bamqc -bam $file -outdir ${file%.bam}_qualimap -outformat HTML -outfile ${file%.bam}_qualimap

done

# Summarize with multiQC
multiqc . --ignore "FastQC/" --ignore "markdup/" --ignore "alignment_stats/"

exit 0

```

Output: PDF reports for each library. Share with class. 

# 4) Call variants

We need to identify all the variants in both the parental and mutant yeast strains. Eventually will compare the list and filter to only look at ones that are in the mutant, but to do that, we need the parental variation too. 

Use bcftools mpileup and bcftools call (v1.22) to get a list of variant sites. 

```{r, eval = F}
 
# This command relies on knowing the ploidy of the yeasts. Samples with 4, 5, 6 in the name are diploid. 
# Move BAMs with *04*, *05*, *06* to a new directory called ploidy2
mv *04*.bam* ploidy2/
mv *05*.bam* ploidy2/  
mv *06*.bam* ploidy2/
# Move the remaining BAMs to a new directory called ploidy1 
mv *.bam* ploidy1/
  
# In the ploidy1 directory, run the following script:   

#!/bin/bash
#SBATCH --job-name=VCF

module load bcftools/1.22

for file in *.bam; do

bcftools mpileup -f Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa $file -o ${file%.bam}.vcf
bcftools call --ploidy 1 -O v -vc ${file%.bam}.vcf -o ${file%.bam}_vars.vcf

done

exit 0

# In the ploidy2 directory, run the following: 

#!/bin/bash
#SBATCH --job-name=VCF

module load bcftools/1.22

for file in *.bam; do

bcftools mpileup -f Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa $file -o ${file%.bam}.vcf
bcftools call --ploidy-file ploidy2_file -O v -vc ${file%.bam}.vcf -o ${file%.bam}_vars.vcf

done

exit 0

# The ploidy2_file contents are: 
#*  * *     * 2

# Link the VCFs in a folder all together again 
ln -s

# Fix the first column so that it isn't just "I" and "II" but is "chrI" and "chrII." 
for file in *.vcf; do
# Starting at the 49th line, append "chr" to the front of each row
awk 'NR>49 {print "chr"$0}' $file > intermediate.vcf
# Add back in the header 
head -n 48 $file > header.temp
cat header.temp intermediate.vcf > ${file%_vars.vcf}.vcf
done

```

Export the variant calls in VCF format. Upload to GitHub. 
